pkg/ocm/helpers_test.go

```diff
@@ -695,3 +695,43 @@ var _ = Describe("GetAutoNodeRoleArn", func() {
 		Expect(exists).To(BeTrue())
 	})
 })
+
+var _ = Describe("ValidateHTTPProxy", func() {
+	It("accepts valid proxy URL with authentication", func() {
+		err := ValidateHTTPProxy("http://user:pass@proxy.example.com:8080")
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("accepts valid proxy URL without authentication", func() {
+		err := ValidateHTTPProxy("http://proxy.example.com:8080")
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("accepts empty string", func() {
+		err := ValidateHTTPProxy("")
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("rejects password with forward slash and provides helpful error", func() {
+		err := ValidateHTTPProxy("http://proxyuser:QvoZjyy/trkCiY5@10.0.0.161:8080")
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("must be URL-encoded"))
+	})
+
+	It("accepts password with URL-encoded forward slash", func() {
+		err := ValidateHTTPProxy("http://proxyuser:QvoZjyy%2FtrkCiY5@10.0.0.161:8080")
```

Comment:
Wondering if system should URL encode it or expect customer to do it

Reply:
I think we should URL encode, `curl` for example does not handle non-URL-encoded characters well on its own (unless they are non-ascii). So in this case, the `@` may cause a failure with `curl` (or another tool, browser, etc)